extensions/gitpod-remote/src/ports.ts

```diff
@@ -0,0 +1,32 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import { GitpodExtensionContext } from 'gitpod-shared';
+import { PortConfig } from '@gitpod/gitpod-protocol/lib/protocol';
+import { PortsStatus, PortsStatusRequest, PortsStatusResponse } from '@gitpod/supervisor-api-grpc/lib/status_pb';
+
+export async function tunnelInitialPorts(context: GitpodExtensionContext, portsConfig: PortConfig[]) {
+	let supervisorPortList: PortsStatus.AsObject[] = [];
+	try {
+		supervisorPortList = await new Promise<PortsStatus.AsObject[]>((resolve, reject) => {
+			const req = new PortsStatusRequest();
+			const evts = context.supervisor.status.portsStatus(req, context.supervisor.metadata);
+			evts.on('error', reject);
+			evts.on('data', (resp: PortsStatusResponse) => resolve(resp.getPortsList().map(p => p.toObject())));
+		});
+	} catch (e) {
+		context.logger.error('Could not fetch ports info from supervisor', e);
+	}
+
+	for (const port of portsConfig) {
+		if (port.onOpen !== 'ignore') {
```

Comment:
if you want to filter out ignored ports then this API should be used: https://github.com/gitpod-io/gitpod/blob/39c5542024f7717041efde80cd1d5476e725ca7b/components/supervisor-api/status.proto#L190